library/posture.c

```diff
@@ -731,6 +731,31 @@ static void process_check_work(uv_work_t *w) {
     pcw->signers = get_signers(path, &pcw->num_signers);
 }
 
+void ziti_endpoint_state_pr_cb(void *unused, const ziti_error *err, void *ctx){
+    if(err) {
+        ZITI_LOG(ERROR, "error during endpoint state posture response submission: %d - %s", err->http_code, err->message);
+    }
+}
+
+
+void ziti_endpoint_state_change(ziti_context ztx, bool woken, bool unlocked) {
+    if(woken || unlocked) {
+        ziti_pr_endpoint_state_req state_req = {
+                .id = "0",
+                .typeId = (char *) PC_ENDPOINT_STATE_TYPE,
+                .unlocked = unlocked,
+                .woken = woken
+        };
+
+        size_t obj_len;
+
+        char *obj = ziti_pr_endpoint_state_req_to_json(&state_req, 0, &obj_len);
```

Comment:
when is this freed?

Reply:
`ziti_pr_post` calls `um_http_req_data` with `obj` as the `body`. The `um_http_req_data` should free it.

Reply:
ok seems that controller(ziti_ctrl.c) frees is (free_body_cb). It is ok, but the preferred way is to free memory in the same layer where it was allocated